tests/dragonfly/__init__.py

```diff
@@ -6,27 +6,40 @@
 import time
 import subprocess
 
+from dataclasses import dataclass
+
+@dataclass
+class DflyParams:
+    path: str
+    cwd: str
+    gdb: bool
+    env: any
 
 class DflyInstance:
     """
     Represents a runnable and stoppable Dragonfly instance
     with fixed arguments.
     """
 
-    def __init__(self, path, args, cwd):
-        self.path = path
+    def __init__(self, params: DflyParams, args):
         self.args = args
-        self.cwd = cwd
+        self.params = params
         self.proc = None
 
     def start(self):
         arglist = DflyInstance.format_args(self.args)
 
         print(f"Starting instance on {self.port} with arguments {arglist}")
-        self.proc = subprocess.Popen([self.path, *arglist], cwd=self.cwd)
+
+        args = [self.params.path, *arglist]
+        if self.params.gdb:
+            args = ["gdb", "--ex", "r", "--args"] + args
```

Comment:
I'm not sure this option is worth it.

Its helpful for me, because in case of a bad signal, the whole process stops. Once brought back to foreground, gdb is directly active and can be used to get stacktrace/dump variables/stack. 

However, it doesn't allow customizing anything before start. Like selecting only a few instances, or setting breakpoints. The latter can be fixed by making the gdb parameter a string and forwarding it to gdb, so just `--gdb` becomes `--gdb="break dflycmd.cc:100; break replica.cc:100"`